packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts

```diff
@@ -763,7 +1019,7 @@ export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGr
         tagSpecifications: tagsToken,
         userData: userDataToken,
         metadataOptions: this.renderMetadataOptions(props),
-        networkInterfaces,
+        networkInterfaces: networkInterfaces ? synthesizeNetworkInterfaces(networkInterfaces) : undefined,
```

Comment:
I think that if `networkInterfaces` is defined, we should not be providing anything in `securityGroupIds`. Since the options provided in `securityGroupsToken` are not used at all unless `props.networkInterfaces` is undefined and `props.associatePublicIpAddress` is defined   
  
On line `securityGroupIds: networkInterfaces ? undefined : securityGroupsToken,` right above